### PR TITLE
taler-depolymerization: init at 0-unstable-2024-06-17

### PR DESCRIPTION
**Pull request by eljamm**
## Description of changes

Taler [depolymerization](https://git.taler.net/depolymerization.git/about/) is an implementation of the Taler wire gateway over the Bitcoin and Ethereum blockchains.

Packaged under the [Summer of Nix](https://github.com/ngi-nix/summer-of-nix) effort.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ta/taler-depolymerization/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    cp -R docs $doc/share/doc/taler-depolymerization
+    cp docs/*.conf $out/share/examples
+  '';
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
+    darwin.apple_sdk.frameworks.Security
+    darwin.apple_sdk.frameworks.SystemConfiguration
+  ];
```
Comment (asymmetric) [inline pkgs/by-name/ta/taler-depolymerization/package.nix:42]: IMO using `with` in these contexts is fine and good

```suggestion
  buildInputs = lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
    CoreFoundation
    Security
    SystemConfiguration
  ]);
```

File: pkgs/by-name/ta/taler-depolymerization/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  meta = {
+    description = "Wire gateway for Bitcoin/Ethereum";
+    homepage = "https://git.taler.net/depolymerization.git/";
+    license = lib.licenses.agpl3Only;
+    maintainers = [
+      # maintained by the team working on NGI-supported software, no group for this yet
```
Comment (asymmetric) [inline pkgs/by-name/ta/taler-depolymerization/package.nix:49]: Maybe would make sense to create it, if you plan on adding yourself in it?

Comment (eljamm) [reply pkgs/by-name/ta/taler-depolymerization/package.nix:49]: I don't think this is a decision I should make on my own. What do you think @fricklerhandwerk?

Comment (fricklerhandwerk) [reply pkgs/by-name/ta/taler-depolymerization/package.nix:52]: Having a team would be good thing to distinguish NGI0 packages and find them programmatically. We don't *really* have the capacity to maintain all of them, but we could try. At worst, the team could be empty simply as a beacon, and we maintain what we can best-effort.

Comment (wegank) [reply pkgs/by-name/ta/taler-depolymerization/package.nix:52]: Yes, that works for me!

**Comment by eljamm** [comment]
Result of `nixpkgs-review pr 342595` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>taler-depolymerization</li>
    <li>taler-depolymerization.doc</li>
  </ul>
</details>

File: pkgs/by-name/ta/taler-depolymerization/package.nix (hunk 2 above)
Comment (eljamm) [reply pkgs/by-name/ta/taler-depolymerization/package.nix:52]: I would also like to join the team, if that's okay.

**Review by asymmetric — APPROVED** [review]

**Pull request merged**